### DISCUSSION TITLE
How to get the analytics of Most searched Keyword from PNP Search box WebParts?

### DISCUSSION THREAD

**Discussion by KTSSangeetha** [microsoft-search/pnp-modern-search#3240] [General]
Describe the bug / error:
How to get the analytics of Most searched Keyword from PNP Search Box WebPart?
We do need the analytics report of Most searched keyword that user has searched for in our PNP search box deployed in SharePoint team site.
We have deployed PNP solutions on SharePoint modern team site to enhance the search experience but unable to get the analytics details of most searched keyword for bringing the appropriate results.

Steps to reproduce:
How to get the analytics of Most searched Keyword from PNP Search Box?
We do need the analytics report of Most searched keyword that user has searched for in our PNP search box deployed in SharePoint team site.
We have deployed PNP solutions on SharePoint modern team site to enhance the search experience but unable to get the analytics details of most searched keyword for bringing the appropriate results.

Expected behavior
How to get the analytics of Most searched Keyword from PNP Search Box?
We do need the analytics report of Most searched keyword that user has searched for in our PNP search box deployed in SharePoint team site.
We have deployed PNP solutions on SharePoint modern team site to enhance the search experience but unable to get the analytics details of most searched keyword for bringing the appropriate results.

**Comment by kasperbolarsen** [comment]
The short version is: you don't
Only the official Microsoft search web parts are logged in the Search & Intelligence logs. It is _possible_ that PnP Modern Search one day might be included as a datasource for search statistics, but it is still just an idea.

**Comment by MoneashKAM** [comment]
Hi @KTSSangeetha and @kasperbolarsen : Greetings! we have a similar requirement to track the searched keywords from PNP search box. Is there any new feature (OOTB) implemented to track it?
or is there any customization possible to track searched keywords from PNP search box to a (SharePoint list)[for ex]

**Reply by kasperbolarsen** [reply]
OOTB there is no way to track that kind of info. As PnP Modern Search is SPFx you can either build a custom extension or you very own version of the web parts. 
I seriously doubt we ever will track search queries as that will deviate from the policy that we don't log ANYTHING.
